tools/rpkg/R/backend-dbplyr__duckdb_connection.R

```diff
@@ -393,5 +393,16 @@ dbplyr_fill0.duckdb_connection <- function(.con, .data, cols_to_fill, order_by_c
   dbplyr_fill0(.con, .data, cols_to_fill, order_by_cols, .direction)
 }
 
+# Customized handling for tbl() to allow the use of scan tables
+# @param src .con A \code{\link{dbConnect}} object, as returned by \code{dbConnect()}
+# @param from Table or parquet/csv -files to be registered
+# @param cache Enable object cache for parquet files
+tbl.duckdb_connection <- function(src, from, cache = FALSE, ...) {
+  ident_q <- pkg_method("ident_q", "dbplyr")
+  if (grepl("^read_parquet|^parquet_scan|\\.parquet'$|^read_csv|\\.csv'", from, ignore.case = TRUE)) from <- ident_q(from)
```

Comment:
I'm not so sure about the name matching business here. Could we not just try to prepare a `SELECT * FROM [from]` and if it succeeds its a table scan? That would also work with replacement scans for data frames in the future for example...

Reply:
Excellent point, thanks! `DBI::dbExistsTable()` seems to be the nicest way to check if a table can be found. Now this works much better, e.g. `duckdb_keywords()` can be used directly with `tbl()`.